src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,930 @@
+    var newCurrent = {
+        prefsSet: event.currentTarget.dataset.prefsset,
+        term: event.currentTarget.dataset.term,
+        //value: event.currentTarget.dataset.value,
+        product: event.currentTarget.dataset.product
+    };
+
+    var newMetadata = {};
+    if (!newCurrent.product) {
+        // Common Term
+        newMetadata.schema = that.options.commonTerms[newCurrent.term];
+        newCurrent.value = that.lookupGenericPrefValue(newCurrent.prefsSet, newCurrent.term);
+    }
+    else {
+        // Application Specific
+        var newProductAppId = newCurrent.product.slice("http://registry.gpii.net/applications/".length);
+        newMetadata = gpii.devpmt.findProductSettingMetadata(that.model.allSolutions,
+            newProductAppId, newCurrent.term);
+        newCurrent.value = that.lookupProductPrefValue(newCurrent.prefsSet, newCurrent.product, newCurrent.term);
+    }
+    newMetadata.name = newMetadata.schema.title;
+    newMetadata.description = newMetadata.schema.description;
+    newCurrent.blank = newCurrent.value === undefined;
+
+    // If blank to false so that you can immediately start editing
+    // the value.
+    newCurrent.blank = false;
+
+    var transaction = that.applier.initiate();
+    transaction.fireChangeRequest({
+        path: "currentlyEditing.current", value: newCurrent
+    });
+    transaction.fireChangeRequest({
+        path: "currentlyEditing.metadata", value: newMetadata
+    });
+    transaction.fireChangeRequest({
+        path: "currentlyEditing.active", value: true
+    });
+    transaction.fireChangeRequest({
+        path: "unsavedChangesExist", value: true
+    });
+    transaction.commit();
+    that.prefsAdjuster.renderInitialMarkup();
+};
+
+
+/**
+ * Setup various bits and pieces of the viewport.
+ */
+gpii.devpmt.editPrefs.initialize = function (that) {
+    // Add sorted solutions
+    var allSolutionsSorted = [];
+    fluid.each(that.model.allSolutions, function (item, key) {
+        var flatItem = fluid.copy(item);
+        flatItem.id = key;
+        allSolutionsSorted.push(flatItem);
+    });
+
+    var sortName = function (a, b) {
+        return a.name.localeCompare(b.name);
+    };
+
+    fluid.stableSort(allSolutionsSorted, sortName);
+    that.applier.change("allSolutionsSorted", allSolutionsSorted);
+
+    // Add sorted commonTerms
+    var commonTermsSorted = [];
+    fluid.each(that.model.commonTerms, function (item, key) {
+        var flatTerm = fluid.copy(item);
+        flatTerm.id = key;
+        commonTermsSorted.push(flatTerm);
+    });
+    that.applier.change("commonTermsSorted", commonTermsSorted);
+
+    fluid.stableSort(that.model.commonTermsSorted, function (a, b) {
+        return a.title.localeCompare(b.title);
+    });
+
+    that.updateCommonTermUsageCounts();
+
+    // Debugging so this component is available in the console
+    var editPrefs = fluid.queryIoCSelector(fluid.rootComponent, "gpii.devpmt.editPrefs")[0];
```

Comment:
Please remove this debugging code - in fact you can find it easily enough using the globalInstantiator

Reply:
Completely Axed.